chainstate/src/detail/mod.rs

```diff
@@ -120,13 +130,15 @@ impl Chainstate {
         chain_config: Arc<ChainConfig>,
         blockchain_storage: blockchain_storage::Store,
         custom_orphan_error_hook: Option<Arc<OrphanErrorHandler>>,
+        custom_time_getter: Option<Arc<TimeGetter>>,
     ) -> Result<Self, crate::ChainstateError> {
         let cons = Self {
             chain_config,
             blockchain_storage,
             orphan_blocks: OrphanBlocksPool::new_default(),
             custom_orphan_error_hook,
             events_controller: EventsController::new(),
+            time_getter: custom_time_getter.unwrap_or_else(|| Arc::new(time::get)),
```

Comment:
If `TimeGetter` was a type wrapper over `Arc<dyn Fn() -> i64 + Send + Sync>`, you could implement `Default` for it and use that (instead of `Some`/`None`) to signify whether the default or a custom time getter is requested.

Reply:
Actually I consider that playing with fire. A default for the `Arc<dyn Fn() -> i64 + Send + Sync>` type doesn't semantically mean anything... by coincidence we have an alias for that type, but by sheer chance it's related to time. On one hand, consider the confusion by some outsider reading the code. Also consider someone importing the wrong Default because of their auto-complete functions in their IDE. Leave alone the case where we, again, by coincidence, we need another function that has the same signature for something else... and guess what default it'll get.

I like the idea of Default in rust, but I consider it one of those slippery slopes that can easily delete the benefit of type-safety. Same issue with From. If a type has a single `Vec<H256>` (think of locators, for example) in it, it doesn't mean that every `Vec<H256>` should be convertible to a locator. Carla almost did that mistake with BlockUndo. That's dangerous territory.

Reply:
Ah, I see. Yeah, that's not bad at all. Let me look into that. Apologies, I misunderstood.

Reply:
I implemented this change. Take a look if you'd like to critique it.